### PR TITLE
use asio for network layer

### PR DESCRIPTION
**Pull request by OznOg**
The pf needs to communicate thru a socket and porting this part is painful on
windows.

This patch uses asio to have a OS agnostic implementation.

Signed-off-by: Sebastien Gonzalve <sebastien.gonzalve@intel.com>
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39138139%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39138166%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39139556%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39152367%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/198%23issuecomment-139236467%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/198%23issuecomment-139236467%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T13%3A36%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2055e44e8af75e2d972742a09ced6c73494fdf934e%20remote-processor/Message.cpp%2018%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39138166%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22idem%22%2C%20%22created_at%22%3A%20%222015-09-10T08%3A55%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/Message.cpp%3AL119-129%22%7D%2C%20%22Pull%2055e44e8af75e2d972742a09ced6c73494fdf934e%20remote-process/main.cpp%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39138139%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%20we%20have%20more%20generic%20socket%20object%20here%20%3F%5Cr%5CnAt%20least%20no%20restriction%20on%20domain%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-09-10T08%3A55%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-process/main.cpp%3AL33-47%22%7D%2C%20%22Pull%2055e44e8af75e2d972742a09ced6c73494fdf934e%20remote-processor/RemoteProcessorServer.cpp%2048%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/198%23discussion_r39139556%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22To%20support%20dual%20stack%20use%20v6%28%29.%22%2C%20%22created_at%22%3A%20%222015-09-10T09%3A13%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T12%3A11%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/RemoteProcessorServer.cpp%3AL39-117%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 55e44e8af75e2d972742a09ced6c73494fdf934e remote-process/main.cpp 9'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/198#discussion_r39138139'>File: remote-process/main.cpp:L33-47</a></b>
- <a href='https://github.com/miguelgaio'></a> Can we have more generic socket object here ?
At least no restriction on domain
- [ ] <a href='#crh-comment-Pull 55e44e8af75e2d972742a09ced6c73494fdf934e remote-processor/Message.cpp 18'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/198#discussion_r39138166'>File: remote-processor/Message.cpp:L119-129</a></b>
- <a href='https://github.com/miguelgaio'></a> idem
- [x] <a href='#crh-comment-Pull 55e44e8af75e2d972742a09ced6c73494fdf934e remote-processor/RemoteProcessorServer.cpp 48'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/198#discussion_r39139556'>File: remote-processor/RemoteProcessorServer.cpp:L39-117</a></b>
- <a href='https://github.com/miguelgaio'></a> To support dual stack use v6().
- <a href='https://github.com/OznOg'></a> done :+1:

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/198?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/198?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/198'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: remote-process/main.cpp [hunk 1]
```diff
@@ -33,15 +33,14 @@
 #include <stdlib.h>
 #include "RequestMessage.h"
 #include "AnswerMessage.h"
-#include "ConnectionSocket.h"
 
 using namespace std;
 
-bool sendAndDisplayCommand(CConnectionSocket &connectionSocket, CRequestMessage &requestMessage)
+bool sendAndDisplayCommand(asio::ip::tcp::socket &socket, CRequestMessage &requestMessage)
```
Comment (miguelgaio) [inline remote-process/main.cpp]: Can we have more generic socket object here ?
At least no restriction on domain

File: remote-processor/Message.cpp [hunk 2]
```diff
@@ -124,9 +119,10 @@ size_t CMessage::getRemainingDataSize() const
 }
 
 // Send/Receive
-CMessage::Result CMessage::serialize(CSocket* pSocket, bool bOut, string& strError)
+CMessage::Result CMessage::serialize(asio::ip::tcp::socket &socket, bool bOut, string& strError)
```
Comment (miguelgaio) [inline remote-processor/Message.cpp]: idem

File: remote-processor/RemoteProcessorServer.cpp [hunk 3]
```diff
@@ -44,112 +39,78 @@
...
-    if (pipe(_aiInbandPipe) == -1) {
-        error = "Could not create a pipe for remote processor communication: ";
-        error += strerror(errno);
-        return false;
-    }
+    using namespace asio;
 
-    // Create server socket
-    std::auto_ptr<CListeningSocket> pListeningSocket(new CListeningSocket);
+    ip::tcp::endpoint endpoint(ip::tcp::v4(), _uiPort);
```
Comment (miguelgaio) [inline remote-processor/RemoteProcessorServer.cpp]: To support dual stack use v6().

Comment (OznOg) [inline remote-processor/RemoteProcessorServer.cpp]: done :+1:

**Comment by krocard** [comment]
:+1:

**Pull request merged**